recipes-kernel/linux/files/0024-net-phy-dp83867-Add-binding-for-the-status-led-confi.patch

```diff
@@ -0,0 +1,154 @@
+index 388ff48f53aee..69b6b0d74c563 100644
+--- a/Documentation/devicetree/bindings/net/ti,dp83867.txt
++++ b/Documentation/devicetree/bindings/net/ti,dp83867.txt
+@@ -42,6 +42,11 @@ Optional property:
+ 				    Some MACs work with differential SGMII clock.
+ 				    See data manual for details.
+ 
++	- ti,led-sel	- This configure the status led. See dt-bindings/net/ti-dp83867.h
++					  for different status led settings,select different configurations
++
++	- ti,led-name   - Correspond name of led source.
++
+ Note: ti,min-output-impedance and ti,max-output-impedance are mutually
+       exclusive. When both properties are present ti,max-output-impedance
+       takes precedence.
+diff --git a/arch/arm64/boot/dts/ti/k3-am65-iot2050-pg2.dtsi b/arch/arm64/boot/dts/ti/k3-am65-iot2050-pg2.dtsi
+index 7972953a41022..17b40a2a6c8d4 100644
+--- a/arch/arm64/boot/dts/ti/k3-am65-iot2050-pg2.dtsi
++++ b/arch/arm64/boot/dts/ti/k3-am65-iot2050-pg2.dtsi
+@@ -797,12 +797,22 @@
+ 		reg = <0>;
+ 		ti,rx-internal-delay = <DP83867_RGMIIDCTL_2_00_NS>;
+ 		ti,fifo-depth = <DP83867_PHYCR_FIFO_DEPTH_4_B_NIB>;
++		ti,led-sel = 	<DP83867_LINK_ESTABLISHED_100BTX>,
++					    <DP83867_LINK_ESTABLISHED_100BTX>,
++						<DP83867_LINK_ESTABLISHED_1000BT>,
++						<DP83867_LINK_ESTABLISHED_BLINK_TRANSMIT_RECEIVE>;
++		ti,led-name = "led_3","led_2","led_1","led_0";
+ 	};
+ 
+ 	pruss0_eth1_phy: ethernet-phy@1 {
+ 		reg = <1>;
+ 		ti,rx-internal-delay = <DP83867_RGMIIDCTL_2_00_NS>;
+ 		ti,fifo-depth = <DP83867_PHYCR_FIFO_DEPTH_4_B_NIB>;
++		ti,led-sel  = 	<DP83867_LINK_ESTABLISHED_100BTX>,
++					    <DP83867_LINK_ESTABLISHED_100BTX>,
++						<DP83867_LINK_ESTABLISHED_1000BT>,
++						<DP83867_LINK_ESTABLISHED_BLINK_TRANSMIT_RECEIVE>;
++		ti,led-name = "led_3","led_2","led_1","led_0";
+ 	};
+ };
+ 
+diff --git a/drivers/net/phy/dp83867.c b/drivers/net/phy/dp83867.c
+index 1b65c8a94e8c3..4b975d59b8bfe 100644
+--- a/drivers/net/phy/dp83867.c
++++ b/drivers/net/phy/dp83867.c
+@@ -4,7 +4,6 @@
+  *
+  * Copyright (C) 2015 Texas Instruments Inc.
+  */
+-
+ #include <linux/ethtool.h>
+ #include <linux/kernel.h>
+ #include <linux/mii.h>
+@@ -44,6 +43,10 @@
+ #define DP83867_10M_SGMII_CFG   0x016F
+ #define DP83867_10M_SGMII_RATE_ADAPT_MASK BIT(7)
+ 
++/* Led Configuration */
++#define DP83867_LEDCR_1      0x0018
++#define LED_GPIO_NUM_SEL	 0x4
++
+ #define DP83867_SW_RESET	BIT(15)
+ #define DP83867_SW_RESTART	BIT(14)
+ 
+@@ -183,6 +186,10 @@ static int dp83867_of_init(struct phy_device *phydev)
+ 	struct device *dev = &phydev->mdio.dev;
+ 	struct device_node *of_node = dev->of_node;
+ 	int ret;
++	u32 led_conf;
++	const char *led_name = NULL;
++	u32 led_select_value;
++	int index;
+ 
+ 	if (!of_node)
+ 		return -ENODEV;
+@@ -287,6 +294,24 @@ static int dp83867_of_init(struct phy_device *phydev)
+ 			   dp83867->fifo_depth);
+ 		return -EINVAL;
+ 	}
++
++	/* Optional LED configuration */
++	for(index = 0;index < LED_GPIO_NUM_SEL;index++) {
```

Comment:
Please make sure to use checkpatch if you are unsure about coding style. This one is off.

Reply:
hi  jan. I have updated the patch,could you pls help to review. thanks

Reply:
OK, I find other bindings referring to defines in headers as well, so that might be fine.

What may happen, though, is that the community will ask for a yaml schema conversion at this chance. Another reason to rush to upstream this so that you may not have to do it or have to rebase if someone else does it meanwhile.